File: Golang/test/maximum_depth_of_binary_tree.go
package test

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func MaximumDepthOfBinaryTree(root *TreeNode) int {
	if root == nil {
		return  0
	}

	depth := 0
	leftDepth := MaximumDepthOfBinaryTree(root.Left)
	rightDepth := MaximumDepthOfBinaryTree(root.Right)

	if leftDepth > rightDepth {
		depth = leftDepth + 1
	}

	depth = rightDepth + 1

	return depth
}

File: Javascript/day_5_recursion_and_dfs_bfs/maximumDepthOfBinaryTree.js
class TreeNode {
  constructor(val, left, right) {
    this.val = val === undefined ? 0 : val;
    this.left = left === undefined ? null : left;
    this.right = right === undefined ? null : right;
  }
}

/**
 * @param {TreeNode} root
 * @return {number}
 */

const root = new TreeNode(10)
root.left = new TreeNode(2)
root.right = new TreeNode(5)

function maximumDepthOfBinaryTree(root) {
  if (!root) {
    return 0;
  }

  const leftDepth = maximumDepthOfBinaryTree(root.left);
  const rightDepth = maximumDepthOfBinaryTree(root.right);

  return Math.max(leftDepth, rightDepth) + 1;
}


console.time()
console.log(maximumDepthOfBinaryTree(root))
console.timeEnd()